tests/unit/syncModel/validation/languageSchema.test.ts

```diff
@@ -0,0 +1,135 @@
+    expect(result.success).toBe(true);
+  });
+
+  test("Valid object with is_default false and valid fallback_language", () => {
+    const validData = {
+      name: "French",
+      codename: "fr",
+      is_active: true,
+      is_default: false,
+      fallback_language: {
+        codename: "en",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(validData);
+    expect(result.success).toBe(true);
+  });
+
+  test("Invalid object with is_default true and fallback_language provided", () => {
+    const invalidData = {
+      name: "English",
+      codename: "en",
+      is_active: true,
+      is_default: true,
+      fallback_language: {
+        codename: "en",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with is_default false and missing fallback_language", () => {
+    const invalidData = {
+      name: "French",
+      codename: "fr",
+      is_active: true,
+      is_default: false,
+      // Missing fallback_language
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object missing required fields", () => {
+    const invalidData = {
+      codename: "en",
+      is_active: true,
+      is_default: true,
+      // Missing 'name'
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with wrong field types", () => {
+    const invalidData = {
+      name: 123, // Should be a string
+      codename: "en",
+      is_active: "true", // Should be a boolean
+      is_default: true,
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with invalid fallback_language", () => {
+    const invalidData = {
+      name: "French",
+      codename: "fr",
+      is_active: true,
+      is_default: false,
+      fallback_language: {
+        // 'codename' is missing
+        code: "en",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with extra properties in fallback_language", () => {
+    const validData = {
+      name: "French",
+      codename: "fr",
+      is_active: true,
+      is_default: false,
+      fallback_language: {
+        codename: "en",
+        extraProp: "ignored",
+      },
+    };
+
+    const result = LanguageSchema.safeParse(validData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object with null properties", () => {
+    const invalidData = {
+      name: null,
+      codename: null,
+      is_active: null,
+      is_default: true,
+    };
+
+    const result = LanguageSchema.safeParse(invalidData);
+    expect(result.success).toBe(false);
+  });
+
+  test("Invalid object that is empty", () => {
```

Comment:
We use `describe/it` functions in this project. Although `test` and `it` behave the same, I would stick to only one of them.